### PR TITLE
[SMALLFIX] Implement isDirectory in Swift UFS

### PR DESCRIPTION
**Pull request by calvinjia**
Valid prefixes ending in a path separator will be counted as "existing" paths.

### CODE REVIEW

**Comment by ifcharming** [comment]
High-level question: is this for a performance improvement?

File: underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java [hunk 1]
```diff
@@ -383,6 +366,19 @@ private boolean copy(String src, String dst) {
   }
 
   /**
+   * Check if the path is a prefix of at least one object in Swift.
```
Comment (apc999) [inline underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java]: NIT: checks

**Comment by ifcharming** [comment]
LGTM

File: underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java [hunk 2]
```diff
@@ -383,6 +366,19 @@ private boolean copy(String src, String dst) {
...
+   * Checks if the path is a prefix of at least one object in Swift.
+   *
+   * @param path the path to check
+   * @return boolean indicating if the path is a directory
+   * @throws IOException if an error occurs listing the directory
+   */
+  private boolean isDirectory(String path) throws IOException {
+    String strippedPath = stripPrefixIfPresent(path);
+    String[] children = listInternal(strippedPath, true);
+    return children != null && children.length > 0;
```
Comment (apc999) [inline underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java]: what if `path` is an empty directory?

Comment (calvinjia) [inline underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java]: This isn't a complete fix, more of a bandaid. The design of the Swift UFS does not really handle directories (ie. `mkdirs` returns true regardless).

Comment (apc999) [inline underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java]: please add a NOTE or TODO here before we merge this PR. thanks

**Comment by apc999** [comment]
LGTM

**Pull request merged**